pkg/cloud/packet/scope/machine.go

```diff
@@ -208,3 +237,160 @@ func (m *MachineScope) GetRawBootstrapData() ([]byte, error) {
 
 	return value, nil
 }
+
+func getProviderIDPrefix(ctx context.Context, mgmtClient client.Client, workloadClientGetter remote.ClusterClientGetter, cluster *clusterv1.Cluster, machine *clusterv1.Machine, packetMachine *infrav1.PacketMachine) (string, error) {
+	// Use existing prefix if already defined
+	if existingPrefix := providerIDPrefixFromPacketMachine(packetMachine); existingPrefix != "" {
+		return existingPrefix, nil
+	}
+
+	// Try to determine the appropriate prefix from any known cloud-provider deployments
+	fromDeployments, err := providerIDFromCloudProviderDeployments(ctx, mgmtClient, workloadClientGetter, cluster)
+	if err != nil {
+		return "", err
+	}
+	if fromDeployments != "" {
+		return fromDeployments, nil
+	}
+
+	// TODO: maybe switch to unstructured?
+	// TODO: add note that this only works for kubeadm bootstrapping, might need
+	// similar support for alternative bootstrappers
+	if machine.Spec.Bootstrap.ConfigRef != nil && machine.Spec.Bootstrap.ConfigRef.Kind == "KubeadmConfig" {
+		kubeadmConfig := new(bootstrapv1.KubeadmConfig)
+		key := client.ObjectKey{
+			Name:      machine.Spec.Bootstrap.ConfigRef.Name,
+			Namespace: packetMachine.Namespace,
+		}
+		// TODO: does this need to take into account notfound?
```

Comment:
It seems if you get a 404 here, you will get an err and skip over the kubeadmConfig dependent blocks.
So it seems you don’t need to do anything 404 sensitive.